File: backend/app/schemas/caregiver.py
from typing import Optional, List
from pydantic import BaseModel, EmailStr, Field

class CaregiverContact(BaseModel):
    name: str = Field(..., json_schema_extra={"example": "Sarah Pendelton (Daughter)"})
    email: EmailStr = Field(..., json_schema_extra={"example": "sarah.caregiver@example.com"})
    phone_number: str = Field(..., json_schema_extra={"example": "+15550192834"})
    fcm_token: Optional[str] = Field(None, json_schema_extra={"example": "fcm_token_sample_abc123"})
    notify_on_missed_dose: bool = True

class CaregiverAlertRequest(BaseModel):
    patient_id: str
    patient_name: str
    medication_name: str
    scheduled_time: str
    minutes_overdue: int

class CaregiverAlertResponse(BaseModel):
    status: str
    alert_id: str
    recipient_count: int
    sent_at: str
    detail: str

File: frontend/lib/models/caregiver.dart
class Caregiver {
  final String id;
  final String name;
  final String email;
  final String phoneNumber;
  final bool notifyOnMissedDose;

  Caregiver({
    required this.id,
    required this.name,
    required this.email,
    required this.phoneNumber,
    required this.notifyOnMissedDose,
  });

  factory Caregiver.fromJson(Map<String, dynamic> json) {
    return Caregiver(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      phoneNumber: json['phone_number'] ?? '',
      notifyOnMissedDose: json['notify_on_missed_dose'] ?? true,
    );
  }
}
